alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/ProviderConfigSelectCustomFunctionAction.java

```diff
@@ -27,47 +27,51 @@
+import com.synopsys.integration.alert.common.rest.HttpServletContentWrapper;
 import com.synopsys.integration.alert.common.rest.model.FieldModel;
+import com.synopsys.integration.alert.common.security.authorization.AuthorizationManager;
 
 @Component
-public class ProviderConfigSelectCustomEndpoint extends SelectCustomEndpoint {
+public class ProviderConfigSelectCustomFunctionAction extends CustomFunctionAction<List<LabelValueSelectOption>> {
     private final ConfigurationAccessor configurationAccessor;
     private final DescriptorMap descriptorMap;
 
     @Autowired
-    public ProviderConfigSelectCustomEndpoint(CustomEndpointManager customEndpointManager, ConfigurationAccessor configurationAccessor, DescriptorMap descriptorMap) throws AlertException {
-        super(ProviderDescriptor.KEY_PROVIDER_CONFIG_NAME, customEndpointManager);
+    public ProviderConfigSelectCustomFunctionAction(AuthorizationManager authorizationManager, ConfigurationAccessor configurationAccessor, DescriptorMap descriptorMap) {
+        super(authorizationManager);
         this.configurationAccessor = configurationAccessor;
         this.descriptorMap = descriptorMap;
     }
 
     @Override
-    protected List<LabelValueSelectOption> createData(FieldModel fieldModel) throws AlertException {
+    public ActionResponse<List<LabelValueSelectOption>> createActionResponse(FieldModel fieldModel, HttpServletContentWrapper servletContentWrapper) throws AlertDatabaseConstraintException {
         String providerName = fieldModel.getDescriptorName();
         Optional<DescriptorKey> descriptorKey = descriptorMap.getDescriptorKey(providerName);
+        List<LabelValueSelectOption> content = List.of();
         if (descriptorKey.isPresent()) {
             List<ConfigurationModel> configurationModels = configurationAccessor.getConfigurationsByDescriptorKeyAndContext(descriptorKey.get(), ConfigContextEnum.GLOBAL);
-            return configurationModels.stream()
-                       .map(ConfigurationModel::getCopyOfKeyToFieldMap)
-                       .map(FieldAccessor::new)
-                       .map(accessor -> accessor.getString(ProviderDescriptor.KEY_PROVIDER_CONFIG_NAME))
-                       .flatMap(Optional::stream)
-                       .map(LabelValueSelectOption::new)
-                       .collect(Collectors.toList());
+            content = configurationModels.stream()
+                          .map(ConfigurationModel::getCopyOfKeyToFieldMap)
+                          .map(FieldAccessor::new)
+                          .map(accessor -> accessor.getString(ProviderDescriptor.KEY_PROVIDER_CONFIG_NAME))
+                          .flatMap(Optional::stream)
+                          .map(LabelValueSelectOption::new)
+                          .collect(Collectors.toList());
         }
-        return List.of();
+        return new ActionResponse<>(HttpStatus.OK, content);
```

Comment:
I'm working backwards here, but we might want to rethink content being a `List`. It adds one more step to the controller for enforcing the pattern of an API always returning a JSON Object.

Reply:
We need to change the corresponding javascript code for those fields to expect an object and rather than an array. I will look into it.

Reply:
Maybe a follow-up PR then, but we could definitely prepare the actions for that change.

Reply:
I have created a new branch [ps_function_select_options](https://github.com/blackducksoftware/blackduck-alert/tree/ps_function_select_options) for the comments because this PR functions with the UI as is.  I will need to make UI changes to update UI fields to render options properly.